backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/attribute/IHUAttributesBL.java

```diff
@@ -56,19 +38,28 @@
 	/**
 	 * Iterates the HU-tree of the given HU and sets the given attribute to the given attributeValue.
 	 *
-	 * @param hu
-	 * @param attribute
-	 * @param attributeValue
 	 * @param onlyHUStatus may be <code>null</code> or empty. Otherwise, only HUs with the given status are updated. However, all HUs are iterated.
+	 * @deprecated
 	 */
+	@Deprecated
 	void updateHUAttributeRecursive(
 			I_M_HU hu,
 			I_M_Attribute attribute,
-			Object attributeValue,
-			String onlyHUStatus);
+			@Nullable Object attributeValue,
+			@Nullable String onlyHUStatus);
+
+	/**
+	 * Iterates the HU-tree of the given HU and sets the given attribute to the given attributeValue.
+	 *
+	 * @param onlyHUStatus may be <code>null</code> or empty. Otherwise, only HUs with the given status are updated. However, all HUs are iterated.
```

Comment:
```suggestion
	 * Iterates the HU-tree of the given HU and sets the given attribute to the given attributeValue.
	 *
	 * Note: for complex scenarios (distributing a weight onto an HU-tree), see {@link de.metas.handlingunits.attribute.propagation.IHUAttributePropagator} and {@link de.metas.handlingunits.attribute.strategy.IAttributeStrategy}.
	 *
	 * @param onlyHUStatus may be <code>null</code> or empty. Otherwise, only HUs with the given status are updated. However, all HUs are iterated.
```